fuse/opcode.go

```diff
@@ -450,7 +450,21 @@ func doStatFs(server *Server, req *request) {
 }
 
 func doIoctl(server *Server, req *request) {
-	req.status = ENOSYS
+	in := (*IoctlIn)(req.inData)
+	out := (*IoctlOut)(req.outData())
+	var bufIn, bufOut []byte
+	if in.InSize > 0 {
+		bufIn = req.arg
+	} else if in.OutSize > 0 {
+		outputsize := unsafe.Sizeof(IoctlOut{})
+		if uintptr(in.OutSize) <= outputHeaderSize-sizeOfOutHeader-outputsize {
+			bufOut = req.outBuf[sizeOfOutHeader+outputsize : sizeOfOutHeader+outputsize+uintptr(in.OutSize)]
```

Comment:
This is nice trick to avoid allocation. 

The number of ioctl calls is expected to be low, I'd like to just call `server.allocOut(req, in.OutSize)` to make it simple.

Reply:
done